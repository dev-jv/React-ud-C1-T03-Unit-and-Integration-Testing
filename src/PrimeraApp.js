import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ color, especie }) => {
        return (
            <>
                <h1> { color } </h1>
                <p> { especie } </p>
            </>
        );
};

PrimeraApp.propTypes = {
    color : PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
    especie: 'Kaede',
};

export default PrimeraApp;



