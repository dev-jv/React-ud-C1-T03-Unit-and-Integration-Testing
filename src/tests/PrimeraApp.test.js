import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';
import "../setupTests"

describe('Pruebas en <PrimeraApp />', () => {

    // ------------------------------------- <> with render...
    // test('Debe definir el color red', () => {
    //     const color = 'Red';
    //     const { getByText } = render(<PrimeraApp color={color}/>);
    //     expect( getByText(color + '') ).toBeInTheDocument();
    //     // const wrapper = render(<PrimeraApp color={color}/>);
    //     // expect( wrapper.getByText(color) ).toBeInTheDocument();
    // })

    // ------------------------------------- <> with Enzyme
    test('Debe mostrar <PrimeraApp/> correctamente', () => {
        const color = 'Red';
        const wrapper = shallow( <PrimeraApp color={ color }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar la especie enviada por props', () => {
        const color = 'Red';
        const especie = 'Miyasama Kaede';
        const wrapper = shallow(
            <PrimeraApp
                color={ color }
                especie={ especie }
            />
        );
        const txtPar = wrapper.find('p').text();
        expect(txtPar).toBe('Miyasama Kaede')
    })
});



