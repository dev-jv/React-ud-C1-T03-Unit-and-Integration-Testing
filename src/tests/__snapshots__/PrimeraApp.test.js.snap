// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Red
  </h1>
  <p>
    Kaede
  </p>
</Fragment>
`;
